libnetwork/drivers/bridge/setup_ip_tables.go

```diff
@@ -396,23 +396,36 @@ func removeIPChains(version iptables.IPVersion) {
 }
 
 func setupInternalNetworkRules(bridgeIface string, addr *net.IPNet, icc, insert bool) error {
-	var (
-		inDropRule  = iptRule{table: iptables.Filter, chain: IsolationChain1, args: []string{"-i", bridgeIface, "!", "-d", addr.String(), "-j", "DROP"}}
-		outDropRule = iptRule{table: iptables.Filter, chain: IsolationChain1, args: []string{"-o", bridgeIface, "!", "-s", addr.String(), "-j", "DROP"}}
-	)
-
 	version := iptables.IPv4
 
-	if addr.IP.To4() == nil {
+	if addr.IP.To4() != nil {
+		var (
+			inDropRule  = iptRule{table: iptables.Filter, chain: IsolationChain1, args: []string{"-i", bridgeIface, "!", "-d", addr.String(), "-j", "DROP"}}
+			outDropRule = iptRule{table: iptables.Filter, chain: IsolationChain1, args: []string{"-o", bridgeIface, "!", "-s", addr.String(), "-j", "DROP"}}
+		)
+		if err := programChainRule(iptables.IPv4, inDropRule, "DROP INCOMING", insert); err != nil {
+			return err
+		}
+		if err := programChainRule(iptables.IPv4, outDropRule, "DROP OUTGOING", insert); err != nil {
+			return err
+		}
+	} else {
 		version = iptables.IPv6
-	}
 
-	if err := programChainRule(version, inDropRule, "DROP INCOMING", insert); err != nil {
-		return err
-	}
-	if err := programChainRule(version, outDropRule, "DROP OUTGOING", insert); err != nil {
-		return err
+		var (
+			inDropRule = iptRule{table: iptables.Filter, chain: IsolationChain1, args: []string{
+				"-i", bridgeIface, "!", "-o", bridgeIface, "!", "-d", addr.String(), "-j", "DROP"}}
+			outDropRule = iptRule{table: iptables.Filter, chain: IsolationChain1, args: []string{
+				"!", "-i", bridgeIface, "-o", bridgeIface, "!", "-s", addr.String(), "-j", "DROP"}}
+		)
+		if err := programChainRule(iptables.IPv4, inDropRule, "DROP INCOMING", insert); err != nil {
+			return err
+		}
+		if err := programChainRule(iptables.IPv4, outDropRule, "DROP OUTGOING", insert); err != nil {
+			return err
+		}
```

Comment:
Shouldn't these be `iptables.IPv6`? (Or `version`?) Why are there separate sets of `programChainRule` calls for v4 vs. v6, but only a single shared `setIcc` call?

Reply:
> Shouldn't these be iptables.IPv6?

Doh :facepalm: Should be fixed now.

> Why are there separate sets of programChainRule calls for v4 vs. v6, but only a single shared setIcc call?

Not sure I get your question. `setIcc` only adds a rule that look like this:

```
-A FORWARD -i <bridge-iface> -o <bridge-iface> -j ACCEPT
```

Hence there's no need to program iptables in different ways.